system/Common.php

```diff
@@ -203,7 +204,12 @@ function command(string $command)
     /**
      * More simple way of getting config instances from Factories
      *
-     * @return object|null
+     * @template ConfigTemplate of BaseConfig
+     *
+     * @param class-string<ConfigTemplate>|string $name
```

Comment:
Is this better according to our current rule?
Probably the type `ConfigTemplate` is not understood by IDEs.

```suggestion
     * @phpstan-param class-string<ConfigTemplate>|string $name
```

Reply:
@kenjis After this change phpstan throws the errors.

Reply:
What errors?
No errors on my local environment.
```diff
--- a/system/Common.php
+++ b/system/Common.php
@@ -206,9 +206,9 @@ if (! function_exists('config')) {
      *
      * @template ConfigTemplate of BaseConfig
      *
-     * @param class-string<ConfigTemplate>|string $name
+     * @phpstan-param class-string<ConfigTemplate>|string $name
      *
-     * @return ConfigTemplate|null
+     * @return object|null
      * @phpstan-return ($name is class-string<ConfigTemplate> ? ConfigTemplate : object|null)
      */
     function config(string $name, bool $getShared = true)
```

Reply:
When we use @return object|null and only @phpstan-param there is no auto suggestion (is not understood by IDEs).

Reply:
@maniaba Exactly, PhpStorm does not do auto suggestion.

Reply:
let's leave it as it is right now?
It is good for phpstan and IDE.